### ISSUE TITLE
Incorrect speed calculation

### ISSUE THREAD

**Issue by ShelleyLake** [BourgeoisLab/GPXLab#29]
@BourgeoisLab @sikmir 
I have a problem with the attached file. Basically all the calculated speeds are wrong.
e.g. points 13 and 14...
How is it that 168.9 metres in about 8 seconds gives a speed of 0 km/h? 
I think it should be 168.9/8*3.6=76Km/h

![grafik](https://user-images.githubusercontent.com/49189962/71577121-31d9d380-2af3-11ea-945e-2dc08374281c.png)

[2009-01-22T21_31_16Z#2#3.zip](https://github.com/BourgeoisLab/GPXLab/files/4009470/2009-01-22T21_31_16Z.2.3.zip)

**Comment by sikmir** [comment]
Multiple time skews detected, e.g.:
```sh
$ sed -n 305,313p 2009-01-22T21_31_16Z\#2\#3.gpx | grep time
      <name>[1.1.0287]</name><time>2009-01-22T21:36:45.214216Z</time></trkpt>
      <name /><time>2009-01-22T21:36:45.164168Z</time></trkpt>
      <name>[1.1.0288]</name><time>2009-01-22T21:36:45.114120Z</time></trkpt>
```

**Comment by BourgeoisLab** [comment]
There was a problem when parsing the timestamps. Fixed with commit 0c710c0.
